rnd/autogpt_builder/src/components/tutorial.ts

```diff
@@ -493,6 +494,15 @@ export const startTutorial = (
     localStorage.setItem("shepherd-tour", "completed"); // Optionally mark the tutorial as completed
   });
 
+  for (const step of tour.steps) {
+    step.on("show", () => {
+      "use client";
```

Comment:
Is this not whole file using the client code ? Can we move the use client up ?

Reply:
The rest of the code doesn’t have to use client as I understand it but this specific block does. My understanding is doing it at the lowest possible level is best for performance so making the whole file client only isn’t advised